src/homeassistant.cpp

```diff
@@ -295,22 +295,25 @@ void HomeAssistant::webSocketSendCommand(const QString &domain, const QString &s
 int HomeAssistant::convertBrightnessToPercentage(float value) { return static_cast<int>(round(value / 255 * 100)); }
 
 void HomeAssistant::updateEntity(const QString &entity_id, const QVariantMap &attr) {
-    EntityInterface *entity = m_entities->getEntityInterface(entity_id);
-    if (entity) {
-        if (entity->type() == "light") {
-            updateLight(entity, attr);
-        }
-        if (entity->type() == "blind") {
-            updateBlind(entity, attr);
-        }
-        if (entity->type() == "media_player") {
-            updateMediaPlayer(entity, attr);
-        }
-        if (entity->type() == "climate") {
-            updateClimate(entity, attr);
-        }
-        if (entity->type() == "switch") {
-            updateSwitch(entity, attr);
+    QList<EntityInterface*> entities = m_entities->getByIntegration(integrationId());
+
+    foreach(EntityInterface *entity, entities) {
+        if (entity->entity_id().startsWith(entity_id)) {
```

Comment:
I'm not that thrilled about parsing all entities until we have a match. Compared to the previous optimized single map access, this is a huge overhead. Especially because it applies to all entity types and the situation gets worse the more entities we have.

Wasn't this change only intended for the remote entity?
If this is the case, then there's no reason to change `updateEntity` since there's no remote entity update. Or I miss something...

FYI: Qt's `foreach` macro is deprecated and will be removed in a future version of Qt. See: https://doc.qt.io/qt-5/qtglobal.html#foreach 
Just be careful when using the new C++11 range-for with Qt containers, so there's no deep copy of the container!